subsys/dsp/cmsis/public/zdsp_backend.h

```diff
@@ -0,0 +1,162 @@
+/* Copyright (c) 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SUBSYS_MATH_CMSIS_BACKEND_PUBLIC_ZDSP_BACKEND_DSP_H_
+#define SUBSYS_MATH_CMSIS_BACKEND_PUBLIC_ZDSP_BACKEND_DSP_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* This include MUST be done before arm_math.h so we can let the arch specific
+ * logic set up the right #define values for arm_math.h
+ */
+#include <zephyr/kernel.h>
+#include <arm_math.h>
+
+#define DEF_ZDSP_MAP(op, type)                                                                     \
+	static inline void zdsp_##op(const type *src_a, const type *src_b, type *dst,              \
+				     uint32_t block_size)                                          \
+	{                                                                                          \
+		arm_##op(src_a, src_b, dst, block_size);                                           \
+	}
+
+DEF_ZDSP_MAP(mult_q7, q7_t)
+DEF_ZDSP_MAP(mult_q15, q15_t)
+DEF_ZDSP_MAP(mult_q31, q31_t)
+DEF_ZDSP_MAP(mult_f32, float32_t)
+
+DEF_ZDSP_MAP(add_q7, q7_t)
+DEF_ZDSP_MAP(add_q15, q15_t)
+DEF_ZDSP_MAP(add_q31, q31_t)
+DEF_ZDSP_MAP(add_f32, float32_t)
+
+DEF_ZDSP_MAP(sub_q7, q7_t)
+DEF_ZDSP_MAP(sub_q15, q15_t)
+DEF_ZDSP_MAP(sub_q31, q31_t)
+DEF_ZDSP_MAP(sub_f32, float32_t)
+
+#undef DEF_ZDSP_MAP
```

Comment:
How about using more descriptive macro name rather than redefining the same one? I always found these patterns a bit harsh to read, could do `DEF_ZDSP_MAP_BASIC`, `DEF_ZDSP_MAP_SCALE`, etc...

Reply:
I went ahead and just expanded them, the macros were really useful to verify that this work, but I think it's more readable without them.